src/lib.rs

```diff
@@ -2188,4 +2259,32 @@ mod tests {
         assert_eq!(result, 10);
         assert_eq!(pool.get_current_worker_count(), 5);
     }
+
+    #[test]
+    fn test_on_callbacks() {
+        let mut pool = ThreadPool::new(1, 50, Duration::from_secs(0));
+        let counter = Arc::new(AtomicUsize::new(0));
+
+        let clone = counter.clone();
+        pool.on_worker_start(move || {
+            clone.fetch_add(1, Ordering::SeqCst);
+        });
+        let clone = counter.clone();
+        pool.on_worker_stop(move || {
+            clone.fetch_sub(1, Ordering::SeqCst);
+        });
+
+        for _ in 0..160 {
+            let clone = counter.clone();
+            pool.execute(move || {
+                let val = clone.load(Ordering::SeqCst);
+                assert!(val > 0);
+            });
+        }
+        thread::sleep(Duration::from_secs(2));
```

Comment:
I'd use the join() function to reduce the chance of race conditions (if it takes longer than 2 seconds for whatever reason). And perhaps even use the on_worker_stop callback to signal a CondVar when the last worker stops and await it with a timeout, instead of sleeping much longer than necessary in most cases.